ui/src/app/pages/login/sign-in.tsx

```diff
@@ -435,7 +441,7 @@ export class SignInImpl extends React.Component<SignInProps, SignInState> {
       if (environment.enableCaptcha) {
         // Reset captcha
         this.captchaRef.current.reset();
-        this.setState({ captchaToken: null, captcha: true });
+        this.setState({ captchaToken: null, captcha: true, loading: false });
```

Comment:
Similarly:
```suggestion
        this.setState(s => { ...s, captchaToken: null, captcha: true, loading: false });
```

Reply:
I am also looking to understand this too. I remember on a recent ticket that you had explained to me a little bit about how state is being handled behind the scenes. Thanks again for that! 

If I implemented the line as you suggest, wouldn't React register a change on each of the attributes, because each of the state attributes are in the object that I am passing, because of the `...s` And is that a problem? Or would it only be a problem if I had something else that was trying to change state concurrently?